lib/mllp/client.ex

```diff
@@ -464,6 +468,34 @@ defmodule MLLP.Client do
     stop_connection(state, reason, "process terminated")
   end
 
+  defp recv_ack(state, opts, buffer) do
+    case state.tcp.recv(state.socket, 0, opts.reply_timeout) do
+      {:ok, reply} ->
+        new_buf = buffer <> reply
+        check = byte_size(new_buf) - 3
+
+        case new_buf do
+          <<@header, _ack::binary-size(check), @trailer>> ->
+            {:ok, new_buf}
+
+          <<@header, _rest::binary>> ->
+            case :binary.split(new_buf, state.mllp_trailer, [:global]) do
+              [_, _ | _] ->
+                {:ok, new_buf}
+
+              _ ->
+                recv_ack(state, opts, new_buf)
+            end
+
+          _ ->
```

Comment:
Here's a rub, do we check that the buffer at least has the header? That is, it may be proceeded by junk, then VT, then some data, but still awaiting the rest.

IMO, we should not, but if that's the case then we should ditch the `:binary.split/3` as well, as then we're being strict, which is fine by me.

Reply:
To be clear, you're in favor of being "strict"? That is, we don't concern ourselves with preceding or trailing junk. The message must be received according to the MLLP spec.

Reply:
Sorry about the confusion. I got confused with header pattern match and your question here.

For Ack to start, we must receive header and check it, right? If ACK is fragmented, rest of the fragment will just append to it until we get trailing char. Sorry maybe I am not understanding the question?  I don't think we need to be concerned about junk, it should still be send as per MLLP spec and if that's what you mean by strict, thats fine.

Reply:
Sounds good to me! :)